src/Services/QuestionnaireService.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace OpenEMR\Services;
+
+use OpenEMR\Common\Uuid\UuidRegistry;
+
+class QuestionnaireService extends BaseService
+{
+    public const TABLE_NAME = 'questionnaire_repository';
+
+    /**
+     * Default constructor.
+     */
+    public function __construct($base_table = null)
+    {
+        parent::__construct($base_table ?? self::TABLE_NAME);
+    }
+
+    public function fetchQuestionnaireResource($name, $q_id = null, $uuid = null)
+    {
+        $sql = "Select * From questionnaire_repository Where (name IS NOT NULL And name = ?) Or (questionnaire_id IS NOT NULL And questionnaire_id = ?)";
+        $bind =  array($name, $q_id);
+        if (!empty($uuid)) {
+            $sql = "Select * From questionnaire_repository Where uuid = ?";
+            $bind =  array($uuid);
+        }
+        $response = sqlQuery($sql, $bind) ?: [];
+        if (is_array($response) && !empty($response['id'])) {
+            $response['uuid'] = UuidRegistry::uuidToString($response['uuid']);
+        }
+        return $response;
+    }
+
+    public function getQuestionnaireIdAndVersion($name, $q_id = null, $uuid = null)
+    {
+        $sql = "Select id, uuid, version From questionnaire_repository Where (name IS NOT NULL And name = ?) Or (questionnaire_id IS NOT NULL And questionnaire_id = ?)";
+        $bind =  array($name, $q_id);
+        if (!empty($uuid)) {
+            $sql = "Select id From questionnaire_repository Where uuid = ?";
+            $bind =  array($uuid);
+        }
+        $response = sqlQuery($sql, $bind) ?: [];
+        if (is_array($response) && !empty($response['id'])) {
+            $response['uuid'] = UuidRegistry::uuidToString($response['uuid']);
+        }
+        return $response;
+    }
+
+    public function fetchQuestionnaireResponses($pid, $id = null, $name = null, $q_id = null)
+    {
+        $sql = "Select * From questionnaire_response Where patient_id = ? And (name = ? Or questionnaire_id = ?)";
+        $resource = sqlStatement($sql, array($pid, $name, $q_id));
+        while ($row = sqlFetchArray($resource)) {
+            $resource[] = $row;
+        }
+        return $resource ?: [];
+    }
+
+    public function saveQuestionnaireResource($q, $name = null, $q_id = null)
+    {
+        $q_version = null;
+        $q_last_date = null;
+        $q_profile = null;
+        $q_status = null;
+        $content = null;
+        $q_ob = null;
+        if (is_string($q)) {
+            $q_ob = json_decode($q, true);
+            $is_json = json_last_error() === JSON_ERROR_NONE;
+            if (!$is_json) {
+                return false;
+            }
+            $q = $q_ob;
+        } else {
+            // in case an object, cast it to array
+            if (is_object($q)) {
+                $q = (array)$q;
+            }
+        }
+        $q_uuid = (new UuidRegistry(['table_name' => 'questionnaire_repository']))->createUuid();
```

Comment:
think this new uuid needs to go into UUID_TABLE_DEFINITIONS in UuidRegistry class